### DISCUSSION TITLE
Idea: change AST nodes from pydantic.BaseModel to pydantic.dataclasses.dataclass

### DISCUSSION THREAD

**Discussion by hacker-DOM** [Ackee-Blockchain/wake#61] [Technical]
It seems that pydantic dataclasses provide all the features of BaseModel, or at least all those we are interested in.

The only disadvantage is that before https://github.com/samuelcolvin/pydantic/pull/4007 is merged, we have to use a workaround[1]

```python3
from pydantic.dataclasses import dataclass
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from dataclasses import dataclass
else:
    from pydantic.dataclasses import dataclass as dataclass
```

The advantages will be fewer auto-complete noise both in type checking and at runtime (ipython). E.g.

```python3
@dataclass
class Dc:
    a: int
    b: int

d=dict(a=1,b=2)
dc=Dc(**d)
```

in iPython, `dc.` has only `a` and `b`.

(On the other hand, using BaseModel, there would be: 
```python3
['a',
 'b',
 'construct',
 'copy',
 'dict',
 'from_orm',
 'json',
 'parse_file',
 'parse_obj',
 'parse_raw',
 'schema',
 'schema_json',
 'update_forward_refs',
 'validate']
```
)

[1] from https://github.com/samuelcolvin/pydantic/issues/4006#issue-1213707998

**Comment by AmaDabla** [comment]
Good idea !
